rules/react.js

```diff
@@ -1,91 +1,151 @@
 		// Allow react to be called and not cause errors.
-		"react/jsx-uses-react": 2,
+		"react/jsx-uses-react": "error",
 		// Allow JSX variables to be marked as used
-		"react/jsx-uses-vars": 2,
-		// Don't allow problematic JSX properties
+		"react/jsx-uses-vars": "error",
+		// Require parentheses around miltline JSX statements.
+		"react/jsx-wrap-multilines": "error",
+		// Warn if you're using array indexes as a key. It might be unavoidable, but use a unique id
+		// if possible.
+		"react/no-array-index-key": "warn",
+		// Don't allow somebody to use "children" as a declared prop.
+		"react/no-children-prop": "error",
+		// Don't allow problematic use of children AND dangerouslySetInnerHTML.
+		"react/no-danger-with-children": "error",
+		// Warn on dangerouslySetInnerHTML -- we should always be aware if we're using it.
 		// More here https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-danger.md
-		"react/no-danger": 0,
-		// Don't allow deprecated React practices
-		"react/no-deprecated": [2, {
-			"react": "0.14.0"
-		}],
+		"react/no-danger": "warn",
+		// Don't allow deprecated React methods.
+		"react/no-deprecated": "error",
 		// Prevent setState in componentDidMount
-		"react/no-did-mount-set-state": [2, "allow-in-func"],
+		"react/no-did-mount-set-state": "error",
 		// Prevent setState in componentDidUpdate
-		"react/no-did-update-set-state": [2, "allow-in-func"],
+		"react/no-did-update-set-state": "error",
 		// Don't allow the state to be mutated. Ever.
-		"react/no-direct-mutation-state": 2,
+		"react/no-direct-mutation-state": "error",
+		// Don't allow findDOMNode, prefer using a callback ref instead.
+		// https://github.com/yannickcr/eslint-plugin-react/issues/678#issue-165177220
+		"react/no-find-dom-node": "error",
 		// Don't allow "isMounted()" calls.
-		"react/no-is-mounted": 2,
+		"react/no-is-mounted": "error",
 		// DISABLING, THERE ARE TIMES WHEN THIS SHOULD BE ALLOWED
 		// Only allow one component to be defined per file.
-		"react/no-multi-comp": 0,
+		"react/no-multi-comp": "off",
+		// Don't use the returned element from rendering with ReactDOM.
+		"react/no-render-return-value": "error",
 		// Allow local state to be changed, even though we're using Redux.
-		"react/no-set-state": 0,
+		"react/no-set-state": "off",
 		// Don't allow string references.
-		"react/no-string-refs": 2,
+		"react/no-string-refs": "error",
+		// Don't allow (usually accidentally) unescaped entities from appearing.
+		"react/no-unescaped-entities": "error",
 		// Don't allow unknown properties.
-		"react/no-unknown-property": 2,
+		"react/no-unknown-property": "error",
+		// Don't allow proptypes to be defined if they aren't used.
+		// This is only off because the caveat leads to numerous false positives. Essentially, all
+		// destructured assignments fail:
+		// https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unused-prop-types.md#caveats
+		"react/no-unused-prop-types": "off",
```

Comment:
@wchest wanted this turned on, but it turns out that the [caveat](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unused-prop-types.md#caveats) is very real and very problematic.

When I tried this on the component library, it failed to recognize pretty much every single prop type that was destructured. It'd be great if it worked, but in reality, it just doesn't.